class Foo(e: Throwable) extends Exception(e)
